src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -580,8 +584,9 @@ object OrderedRVD {
         partitioner.sparkPartitioner(rdd.sparkContext))
         .setKeyOrdering(typ.kOrd)
         .mapPartitionsWithIndex { case (i, it) =>
+          val localPartitioner = partitionerBc.value
           it.map { case (k, v) =>
-            assert(partitioner.getPartition(k) == i)
+            assert(localPartitioner.getPartition(k) == i)
```

Comment:
This will have to stay: if you keep the partitioner, you'll have to call OrderedRVDPartitioner.broadcast and use that in the mapPartitions step.